src/cluster_legacy.c

```diff
@@ -2815,6 +2869,61 @@ static clusterNode *getNodeFromLinkAndMsg(clusterLink *link, clusterMsg *hdr) {
     return sender;
 }
 
+static clusterMsgDataPublishMessage *getFirstPublishMessage(clusterMsgDataPublishMulti *msg) {
+    clusterMsgDataPublishMessage *initial = (clusterMsgDataPublishMessage *)&(msg->bulk_data);
+    return initial;
+}
+
+static clusterMsgDataPublishMessage *getPublishMessageNext(clusterMsgDataPublishMessage *message_cursor) {
+    clusterMsgDataPublishMessage *next =
+        (clusterMsgDataPublishMessage *)((char *)message_cursor->message_data + ntohl(message_cursor->message_len));
+    return next;
```

Comment:
nit - less typing and reading
```suggestion
    return (clusterMsgDataPublishMessage *)((char *)message_cursor->message_data + ntohl(message_cursor->message_len));
```